app/src/main/java/net/squanchy/navigation/deeplink/DeepLinkRouter.java

```diff
@@ -0,0 +1,54 @@
+package net.squanchy.navigation.deeplink;
+
+import android.content.Intent;
+
+import java.net.URI;
+import java.util.regex.Pattern;
+
+public class DeepLinkRouter {
+
+    private static final Pattern PATTERN_VALID_SCHEME = Pattern.compile("[^\\w]");
+
+    private final String scheme;
+    private final NavigationStrategyFactory navigationStrategyFactory;
+    private final DeepLinkNavigator navigator;
+
+    DeepLinkRouter(String scheme, NavigationStrategyFactory navigationStrategyFactory, DeepLinkNavigator navigator) {
+        this.navigator = navigator;
+        validateScheme(scheme);
+        this.scheme = scheme;
+        this.navigationStrategyFactory = navigationStrategyFactory;
+    }
+
+    private void validateScheme(String scheme) {
+        if (PATTERN_VALID_SCHEME.matcher(scheme).find()) {
```

Comment:
isn't this meaning the pattern is for an invalid scheme (read as: you're trying to detect invalid schemes through this pattern)?